src/app/App.js

```diff
@@ -40,6 +42,11 @@ function renderMainContainer(parent) {
   parent.appendChild(comp);
 }
 
+function renderBtn(parent, btnObj = {id: undefined, btnText: 'Hall of fame', class: [], onClickFn: undefined, icon: 'fame'}) {
```

Comment:
Proponuję w końcu uspójnić funkcję render i usunać zbędne powtórzenia :) 
Mamy zawsze taki sam schemat, więc można użyć funkcji:
```
function render({component, parent, wrapperClass}){
  if(wrapperClass){
    component.classList.add(wrapperClass);
  }
  parent.appendChild(component)
  return component;
}

```
I wtedy możemy tak:
```
  const mainContainer = render({
    component: MainContainer(),
    parent: wrapper,
    wrapperClass: 'wrapper__mainContainer'
  })
  const hallOfFameButton = render({
    component: Btn({id: undefined, btnText: 'Hall of fame', class: [], onClickFn: undefined, icon: 'fame'}),
    parent: mainContainer
  })
```

Co Wy na to :) ? 
Dodatkowo nie musimy wyszukiwac obiektu po id (bo tak to id mamy np. w MainContainer a potem niejako dobieramy sie do wnętrznosci mainContainer wyszukujac je po document.getElementById).

Reply:
Druga opcja to jest jeszcze z kompozycją 2 funkcji (bo tak to nam mogą rosnąć parametry, tutaj funkcja dodaje wrapper dodatkowo):
```
export const App = ({ options }) => {
  /** Tutaj reszta **/
  const mainContainer = wrapped({
    component: render({
      component: MainContainer(),
      parent: wrapper,
    }),
    wrapperClass: 'wrapper__mainContainer',
  });
  const hallOfFameButton = render({
    component: Btn({ id: undefined, btnText: 'Hall of fame', class: [], onClickFn: undefined, icon: 'fame' }),
    parent: mainContainer,
  });
};

function render ({ component, parent }) {
  parent.appendChild(component);
  return component;
}

function wrapped ({ component, wrapperClass }) {
  component.classList.add(wrapperClass);
  return component;
}
```

Chociaż w tym wypadku skłaniam się ku 1 opcji, ponieważ ten wrapper mamy praktycznie zawsze.

Reply:
No właśnie też mi się ten mały bajzel nie podoba w App, zrobię porządek :)

Reply:
Jeśli będziesz robił z tym porządek, to może stwórz niezależną osobną gałąź.